### ISSUE TITLE
Wrong expression

### ISSUE THREAD

**Issue by Roushelfy** [MuGdxy/SymEigen#2]
```
A= Eigen.Matrix('A', 2, 2)
B= Eigen.Matrix('B', 2, 2)
AB=A*B
AB2=(AB.transpose()*AB).trace()
print(Closure('AB2', AB2, 'AB2'))

template <typename T>
__device__ __host__ void AB2(T& AB2, const T& Mu, const T& Lambda, const Eigen::Vector<T,6>& X, const Eigen::Matrix<T,2,2>& IB, const T& vol)
{
/*****************************************************************************************************************************
Function generated by SymEigen.py 
Author: MuGdxy
GitHub: https://github.com/MuGdxy/SymEigen
E-Mail: lxy819469559@gmail.com
******************************************************************************************************************************
LaTeX expression:
//tex:$$AB2 = \left(A(0,0) B(0,0) + A(0,1) B(1,0)\right)^{2} + \left(A(0,0) B(0,1) + A(0,1) B(1,1)\right)^{2} + \left(A(1,0) B(0,0) + A(1,1) B(1,0)\right)^{2} + \left(A(1,0) B(0,1) + A(1,1) B(1,1)\right)^{2}$$

Symbol Name Mapping:
Mu:
    -> {}
    -> Matrix([[Mu]])
Lambda:
    -> {}
    -> Matrix([[Lambda]])
X:
    -> {}
    -> Matrix([[X(0)], [X(1)], [X(2)], [X(3)], [X(4)], [X(5)]])
IB:
    -> {}
    -> Matrix([[IB(0,0), IB(0,1)], [IB(1,0), IB(1,1)]])
vol:
    -> {}
    -> Matrix([[vol]])
*****************************************************************************************************************************/
/* Sub Exprs */
/* Simplified Expr */
AB2 = (A(0,0)*B(0,0) + A(0,1)*B(1,0) * A(0,0)*B(0,0) + A(0,1)*B(1,0)) + (A(0,0)*B(0,1) + A(0,1)*B(1,1) * A(0,0)*B(0,1) + A(0,1)*B(1,1)) + (A(1,0)*B(0,0) + A(1,1)*B(1,0) * A(1,0)*B(0,0) + A(1,1)*B(1,0)) + (A(1,0)*B(0,1) + A(1,1)*B(1,1) * A(1,0)*B(0,1) + A(1,1)*B(1,1));
}
```

**Comment by Roushelfy** [comment]
```
a simpler one：
Expression = (Mu+Lambda)**2
print(Closure('Expression', Expression, 'Expression'))
....
/* Simplified Expr */
Expression = (Lambda + Mu * Lambda + Mu);
....
```

**Comment by MuGdxy** [comment]
fixed 748e17c13ebd8bceddceb9f3082978c1dd4ac07d

[Code](https://github.com/MuGdxy/SymEigen/blob/main/test/issue2.ipynb)
